### DISCUSSION TITLE
encrypting a bundle

### DISCUSSION THREAD

**Discussion by ayrley** [rauc/rauc#1541] [General]
Hi

I am trying to encrypt my bundle, unfortunately this fails with the message "No keyring file or directory provided"
The debug flag does not give me a lot more info...

Any help on this?
I added below rauc commands.

`rauc bundle --cert=rauc-keyring/keyring_cert.pem --key=rauc-keyring/keyring_key.pem $PKGDIR $OUTPUTFILE`
`echo "Encrypting bundle"`
`rauc -d encrypt --to=recipients_cert.pem $OUTPUTFILE $OUTPUTFILE.enc.raucb`

**Comment by ejoerns** [comment]
You need to add `--keyring=<keyring.pem>` to your `rauc encrypt` call to verify the bundle before encrypting it. The documentation and help text could maybe be more specific here.

**Reply by ayrley** [reply]
Hi

I do not have the keyring file?
I signed my bundle with a the `--key` and `--cert` properies.

The documentation is not clear at all.
Nothing is mentioned about this when encrypting the bundle. The `--help` also does not mention this.

**Reply by ejoerns** [reply]
@ayrley I bet you'll need a keyring for verification on the target anyway... :smirk: 
Just provide that one.

We're always happy to accept patches for the documentation.
